### ISSUE TITLE
All predictions are the same

### ISSUE THREAD

**Issue by zygmuntz** [tmadl/sklearn-expertsys#3]
When I run the hepatitis demo, I get a nice list of rules:

```
Trained RuleListClassifier for detecting survival
==================================================
IF ALBUMIN : -inf_to_2.65 THEN probability of survival: 87.5% (59.0%-99.6%)
ELSE IF BILIRUBIN : 1.41375838926_to_inf THEN probability of survival: 36.8% (22.5%-52.5%)

ELSE IF ALBUMIN : 3.85863309352_to_inf THEN probability of survival: 3.6% (0.4%-9.7%)
ELSE probability of survival: 18.2% (5.4%-36.3%)
=================================================
```

Reported accuracy is 76.9% and this is correct, but all predictions are the same:

```
array([[ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818],
       [ 0.81818182,  0.18181818]])
```

Similarly, when I run on my dataset, there's a nice list of rules, but all predictions are the same (0).

**Comment by tmadl** [comment]
Hi, I can't reproduce this error. `clf.predict_proba(Xtest)` gives some decent probabilities on my end (see below). Can you let me know what exactly you did to produce your predictions, and send me your version of scikit-learn?

`[[ 0.68181818  0.31818182]
 [ 0.96774194  0.03225806]
 [ 0.96774194  0.03225806]
 [ 0.68181818  0.31818182]
 [ 0.38235294  0.61764706]
 [ 0.38235294  0.61764706]
 [ 0.96774194  0.03225806]
 [ 0.68181818  0.31818182]
 [ 0.96774194  0.03225806]
 [ 0.38235294  0.61764706]
 [ 0.68181818  0.31818182]
 [ 0.68181818  0.31818182]
 [ 0.96774194  0.03225806]
 [ 0.68181818  0.31818182]
 [ 0.96774194  0.03225806]
 [ 0.96774194  0.03225806]
 [ 0.96774194  0.03225806]
 [ 0.96774194  0.03225806]
 [ 0.96774194  0.03225806]
 [ 0.38235294  0.61764706]
 [ 0.38235294  0.61764706]
 [ 0.96774194  0.03225806]
 [ 0.96774194  0.03225806]
 [ 0.96774194  0.03225806]
 [ 0.96774194  0.03225806]
 [ 0.38235294  0.61764706]
 [ 0.96774194  0.03225806]
 [ 0.68181818  0.31818182]
 [ 0.38235294  0.61764706]
 [ 0.96774194  0.03225806]
 [ 0.96774194  0.03225806]
 [ 0.96774194  0.03225806]
 [ 0.96774194  0.03225806]
 [ 0.96774194  0.03225806]
 [ 0.96774194  0.03225806]
 [ 0.68181818  0.31818182]
 [ 0.68181818  0.31818182]
 [ 0.96774194  0.03225806]
 [ 0.96774194  0.03225806]]`

**Comment by tmadl** [comment]
I did a fresh install of up-to-date libraries on both Ubuntu and Windows 7 and still cannot reproduce the error.
